### PR TITLE
[🐛] fix for special characters on file name

### PR DESCRIPTION
**Pull request by buster95**
I'm using NextJS that forces me to use `[ ]` on file names, but when I try to run this files throws errors, I realized is something related with string scape so this PR was created with that purpose

image

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    return ['^', s, '$'].join('');
+}
+
+export function escapeSingleQuotes(s: string): string {
+    return isWindows() ? s : s.replace(/'/g, "'\\''");
+}
+
+export function quote(s: string): string {
+    const q = isWindows() ? '"' : `'`;
+    return [q, s, q].join('');
```
Comment (buberdds) [inline src/util.ts]: Let's use a template string

**Review by buberdds — COMMENTED** [review]
Thanks for looking at this. Can you please clean up utils a little bit? There are functions that are not used by extension.

File: src/util.ts (hunk 1 above)
Comment (buster95) [reply src/util.ts]: it was changed

File: src/util.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+}
+
+export function resolveTestNameStringInterpolation(s: string): string {
+    const variableRegex = /(\${?[A-Za-z0-9_]+}?|%[psdifjo#%])/gi;
+    const matchAny = '(.*?)';
+    return s.replace(variableRegex, matchAny);
+}
+
+export function escapeSingleQuotes(s: string): string {
```
Comment (buberdds) [inline src/util.ts]: not used

File: src/util.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+export function normalizePath(path: string): string {
+    return isWindows() ? path.replace(/\\/g, '/') : path;
+}
+
+export function escapeRegExpForPath(s: string): string {
+    return s.replace(/[*+?^${}<>()|[\]]/g, '\\$&'); // $& means the whole matched string
+}
+
+export function findFullTestName(selectedLine: number, children: any[]): string | undefined {
```
Comment (buberdds) [inline src/util.ts]: not used

File: src/util.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    for (const element of children) {
+        const result = findFullTestName(selectedLine, element.children);
+        if (result) {
+            return resolveTestNameStringInterpolation(element.name) + ' ' + result;
+        }
+    }
+}
+
+export function resolveTestNameStringInterpolation(s: string): string {
```
Comment (buberdds) [inline src/util.ts]: won't be neede dafter removal of `findFullTestName`

**Comment by buster95** [comment]
I think is ready!

**Comment by buberdds** [comment]
Thank you !

**Pull request merged**